GVRf/Extensions/3DCursor/IODevices/gearwear/PhoneSide/gearinputprovider/src/main/java/com/samsung/mpl/gearinputprovider/backend/InputProviderService.java

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (c) 2016. Samsung Electronics Co., LTD
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License..
+ */
+
+package com.samsung.mpl.gearinputprovider.backend;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.IBinder;
+import android.os.Message;
+import android.os.Messenger;
+import android.os.Parcelable;
+import android.os.RemoteException;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.gson.Gson;
+import com.samsung.android.sdk.SsdkUnsupportedException;
+import com.samsung.android.sdk.accessory.SA;
+import com.samsung.android.sdk.accessory.SAAgent;
+import com.samsung.android.sdk.accessory.SAPeerAgent;
+import com.samsung.android.sdk.accessory.SASocket;
+import com.samsung.mpl.gearinputprovider.R;
+import com.samsung.mpl.gearinputprovider.models.NetworkMessage;
+import com.samsung.mpl.gearinputprovider.models.NetworkMessage.Event;
+import com.samsung.mpl.gearinputprovider.models.NetworkMessage.Type;
+import com.samsung.mpl.gearinputprovider.models.NotificationModel;
+import com.samsung.mpl.gearinputprovider.utils.NotificationUtility;
+import com.samsung.mpl.gearinputprovider.utils.Utility;
+import com.samsung.mpl.gearwearlibrary.EventManager;
+import com.samsung.mpl.gearwearlibrary.models.events.Back;
+import com.samsung.mpl.gearwearlibrary.models.events.Click;
+import com.samsung.mpl.gearwearlibrary.models.events.Connected;
+import com.samsung.mpl.gearwearlibrary.models.events.Disconnected;
+import com.samsung.mpl.gearwearlibrary.models.events.Rotary;
+import com.samsung.mpl.gearwearlibrary.models.events.Swipe;
+import com.samsung.mpl.gearwearlibrary.models.events.TouchEnd;
+import com.samsung.mpl.gearwearlibrary.models.events.TouchMove;
+import com.samsung.mpl.gearwearlibrary.models.events.TouchStart;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+
+public class InputProviderService extends SAAgent {
+    private static final String TAG = InputProviderService.class.getSimpleName();
+    private static final String HELLO_WORLD_LINKED_PACKAGE_NAME = "com.samsung.smcl" +
```

Comment:
removed that part of the code all together. It was was two way communication, and currently only one way communication (from watch to phone) is used.